### PR TITLE
First time retention

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
*If this affects the front-end, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Organization, Team, and User (if this PR affects ANY querysets/queries).
- [ ] Django backend tests (if this PR affects the backend).
- [ ] Cypress end-to-end tests (if this PR affects the frontend).

### CODE REVIEW

**Comment by timgl** [comment]
Merge conflict

File: ee/clickhouse/queries/clickhouse_retention.py [hunk 1]
```diff
@@ -100,6 +87,56 @@ def calculate_retention(self, filter: Filter, team: Team, total_intervals: int)
...
+        )
+
+        target_query, params = self._get_condition(target_entity)
+        returning_query, returning_params = self._get_condition(
+            returning_entity, "returning" if first_time_retention else ""
+        )
+
+        query = "AND {target_query}".format(target_query=target_query)
+        returning_query = (
+            "AND ({target_query} OR {returning_query})".format(
```
Comment (EDsCODE) [inline ee/clickhouse/queries/clickhouse_retention.py:119]: I count the first event as part of the return conditions to make sure the first column of events is wrapped in but I think this is based on bad assumptions that people will have clear cut first time events. Going to revisit this tomorrow and readjust this

**Comment by EDsCODE** [comment]
also wait on #2241 as it will be much more easily extendable after

**Comment by EDsCODE** [comment]
closing in favor of #2325

**Pull request closed without merging**